Sources/Impl/PinLayoutImpl.swift

```diff
@@ -818,47 +818,99 @@ class PinLayoutImpl: PinLayout {
         return self
     }
 
+    @discardableResult
+    func marginTop(_ percent: Percent) -> PinLayout {
+        func context() -> String { return "marginTop(\(percent.description))" }
+        guard let layoutSuperviewRect = layoutSuperviewRect(context) else { return self }
+        marginTop = percent.of(layoutSuperviewRect.height)
+        return self
+    }
+
     @discardableResult
     func marginLeft(_ value: CGFloat) -> PinLayout {
         marginLeft = value
         return self
     }
 
+    @discardableResult
+    func marginLeft(_ percent: Percent) -> PinLayout {
+        func context() -> String { return "marginLeft(\(percent.description))" }
+        guard let layoutSuperviewRect = layoutSuperviewRect(context) else { return self }
+        marginLeft = percent.of(layoutSuperviewRect.width)
+        return self
+    }
+
     @discardableResult
     func marginBottom(_ value: CGFloat) -> PinLayout {
         marginBottom = value
         return self
     }
 
+    @discardableResult
+    func marginBottom(_ percent: Percent) -> PinLayout {
+        func context() -> String { return "marginBottom(\(percent.description))" }
+        guard let layoutSuperviewRect = layoutSuperviewRect(context) else { return self }
+        marginBottom = percent.of(layoutSuperviewRect.height)
+        return self
+    }
+
     @discardableResult
     func marginRight(_ value: CGFloat) -> PinLayout {
         marginRight = value
         return self
     }
+
+    @discardableResult
+    func marginRight(_ percent: Percent) -> PinLayout {
+        func context() -> String { return "marginRight(\(percent.description))" }
+        guard let layoutSuperviewRect = layoutSuperviewRect(context) else { return self }
+        marginRight = percent.of(layoutSuperviewRect.width)
+        return self
+    }
     
     @discardableResult
     func marginStart(_ value: CGFloat) -> PinLayout {
         return isLTR() ? marginLeft(value) : marginRight(value)
     }
+
+    @discardableResult
+    func marginStart(_ percent: Percent) -> PinLayout {
+        return isLTR() ? marginLeft(percent) : marginRight(percent)
+    }
     
     @discardableResult
     func marginEnd(_ value: CGFloat) -> PinLayout {
         return isLTR() ? marginRight(value) : marginLeft(value)
     }
 
+    @discardableResult
+    func marginEnd(_ percent: Percent) -> PinLayout {
+        return isLTR() ? marginRight(percent) : marginLeft(percent)
```

Comment:
Here if there is a warning, the message will display the wrong context (either marginRight or marginLeft), it should display "marginEnd".